src/include/txn_inline.h

```diff
@@ -909,6 +910,16 @@ __wt_upd_alloc(WT_SESSION_IMPL *session, const WT_ITEM *value, u_int modify_type
         (value != NULL &&
           !(modify_type == WT_UPDATE_RESERVE || modify_type == WT_UPDATE_TOMBSTONE)));
 
+    /*
+     * If there is no value ensure that the memory allocation size is matches that returned by
+     * sizeof(). Otherwise bit-exact tools like MSan may infer the structure is not completely
```

Comment:
Could you move this comment into `btmem.h` where the define is. I *think* this code would be OK without a comment, given the names we've ended up with, but feel free to leave one here if you think it's necessary.

Reply:
Moved comment. Did not add new comment in this code path.